src/common/AdaptiveDecrypter.h

```diff
@@ -43,7 +36,31 @@ class Adaptive_CencSingleSampleDecrypter : public AP4_CencSingleSampleDecrypter
     throw std::logic_error("SetDefaultKeyId method not implemented.");
   };
 
-protected:
-  uint32_t m_CryptBlocks{0};
-  uint32_t m_SkipBlocks{0};
+  virtual AP4_Result SetFragmentInfo(AP4_UI32 pool_id,
+                                     const AP4_UI08* key,
+                                     const AP4_UI08 nal_length_size,
+                                     AP4_DataBuffer& annexb_sps_pps,
+                                     AP4_UI32 flags,
+                                     AP4_UI08 cryptoType,
+                                     AP4_UI08 cryptBlocks,
+                                     AP4_UI08 skipBlocks)
```

Comment:
since we define an our method we can use `CryptoMode` to remove all casts?

```suggestion
  virtual AP4_Result SetFragmentInfo(AP4_UI32 pool_id,
                                     const AP4_UI08* key,
                                     const AP4_UI08 nal_length_size,
                                     AP4_DataBuffer& annexb_sps_pps,
                                     AP4_UI32 flags,
                                     CryptoMode cryptoMode,
                                     AP4_UI08 cryptBlocks,
                                     AP4_UI08 skipBlocks)
```

Reply:
I've gone one better and now send the whole `CryptoInfo` struct. This should have been simple but there were circular includes which took some time to sort out.